src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ClientCertificates.cs

```diff
@@ -104,69 +97,81 @@ public async Task Manual_SSLBackendNotSupported_ThrowsPlatformNotSupportedExcept
-            }
-            else if (certIndex == 3)
-            {
-                // This is an invalid client cert since it has an EKU but is missing ClientAuthentication OID.
-                clientCert = Configuration.Certificates.GetServerCertificate();
-            }
-
-            Assert.NotNull(clientCert);
+            var options = new LoopbackServer.Options { UseSsl = true };
 
-            HttpClientHandler handler = CreateHttpClientHandler(useSocketsHttpHandlerString, useHttp2String);
-            handler.ClientCertificates.Add(clientCert);
-            using (HttpClient client = CreateHttpClient(handler, useHttp2String))
+            X509Certificate2 GetClientCertificate(int certIndex)
             {
-                var request = new HttpRequestMessage();
-                request.RequestUri = new Uri(Configuration.Http.EchoClientCertificateRemoteServer);
-
-                // Issue #35239. Force HTTP/1.1.
-                request.Version = new Version(1,1);
-                HttpResponseMessage response = client.SendAsync(request).GetAwaiter().GetResult(); // need a 4-arg overload of RemoteInvoke that returns a Task
-                Assert.Equal(expectedStatusCode, response.StatusCode);
+                X509Certificate2 cert = null;
 
-                if (expectedStatusCode == HttpStatusCode.OK)
+                // Get client certificate. RemoteInvoke doesn't allow easy marshaling of complex types.
+                // So we have to select the certificate at this point in the test execution.
+                if (certIndex == 1)
                 {
-                    string body = response.Content.ReadAsStringAsync().GetAwaiter().GetResult(); // need a 4-arg overload of RemoteInvoke that returns a Task
-                    byte[] bytes = Convert.FromBase64String(body);
-                    var receivedCert = new X509Certificate2(bytes);
-                    Assert.Equal(clientCert, receivedCert);
+                    // This is a valid client cert since it has an EKU with a ClientAuthentication OID.
+                    cert = Configuration.Certificates.GetClientCertificate();
                 }
+                else if (certIndex == 2)
+                {
+                    // This is a valid client cert since it has no EKU thus all usages are permitted.
+                    cert = Configuration.Certificates.GetNoEKUCertificate();
+                }
+                else if (certIndex == 3)
+                {
+                    // This is an invalid client cert since it has an EKU but is missing ClientAuthentication OID.
+                    cert = Configuration.Certificates.GetServerCertificate();
+                }
+
+                return cert;
             }
+
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (X509Certificate2 cert = GetClientCertificate(certIndex))
+                using (HttpClient client = CreateHttpClientWithCert(cert))
```

Comment:
Nit: either indent everything that's in the body of these using, or use the using declaration syntax:
```C#
using X509Certificate2 cert = GetClientCertificate(certIndex);
using HttpClient client = CreateHttpClientWithCert(cert);
```

Reply:
Thanks.  I had thought that my syntax was equivalent. But I'll switch as you suggest.

Reply:
> I had thought that my syntax was equivalent

It's not.  The syntax you had means that it's a normal using, which means the only thing that'll be in scope of the using is the next statement, either a block if braces are used or literally the next statement.  The syntax above makes the body of the using everything until the end of the containing scope.

Reply:
Interesting. The test still worked with my (incorrect) syntax. Thanks for pointing this out.

Reply:
Yup, because the next statement was the only one :)